src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -256,7 +256,8 @@ def _resolve_role_id(role, scope, definitions_client):
     role_id = None
     try:
         uuid.UUID(role)
-        role_id = role
+        role_id = '/subscriptions/{}/providers/Microsoft.Authorization/roleDefinitions/{}'.format(
+            definitions_client.config.subscription_id, role)
```

Comment:
Note, role definition is not a resource id, so i can't do exactly like you suggested, rather I will match the pattern through re.